pkg/operator/controller/externaldns/deployment_test.go

```diff
@@ -2873,14 +2875,15 @@ func TestExternalDNSDeploymentChanged(t *testing.T) {
 			description: "if externalDNS has extra container",
 			originalDeployment: testDeploymentWithContainers([]corev1.Container{
 				testContainer(),
-				testContainerWithName("second"),
+				testContainerWithName("extra"),
 			}),
 			mutate: func(depl *appsv1.Deployment) {
 				depl.Spec.Template.Spec.Containers = []corev1.Container{testContainer()}
 			},
-			expect: true,
+			expect: false,
 			expectedDeployment: testDeploymentWithContainers([]corev1.Container{
 				testContainer(),
+				testContainerWithName("extra"),
```

Comment:
If `expect` is false, why specify `expectedDeployment`?

Reply:
No reason, removed it in all the test cases to avoid confusion.